src/api/IBMi.ts

```diff
@@ -944,8 +863,150 @@ export default class IBMi {
+              this.qccsid = cachedServerSettings.qccsid;
+              this.userJobCcsid = cachedServerSettings.jobCcsid;
+              this.userDefaultCCSID = cachedServerSettings.userDefaultCCSID;
+            } else {
+              progress.report({
+                message: `Fetching conversion values.`
+              });
+
+              // Next, we're going to see if we can get the CCSID from the user or the system.
+              // Some things don't work without it!!!
+              try {
+
+                // we need to grab the system CCSID (QCCSID)
+                const [systemCCSID] = await this.runSQL(`select SYSTEM_VALUE_NAME, CURRENT_NUMERIC_VALUE from QSYS2.SYSTEM_VALUE_INFO where SYSTEM_VALUE_NAME = 'QCCSID'`);
+                if (typeof systemCCSID.CURRENT_NUMERIC_VALUE === 'number') {
+                  this.qccsid = systemCCSID.CURRENT_NUMERIC_VALUE;
+                }
+
+                // we grab the users default CCSID
+                const [userInfo] = await this.runSQL(`select CHARACTER_CODE_SET_ID from table( QSYS2.QSYUSRINFO( USERNAME => upper('${this.currentUser}') ) )`);
+                if (userInfo.CHARACTER_CODE_SET_ID !== `null` && typeof userInfo.CHARACTER_CODE_SET_ID === 'number') {
+                  this.userJobCcsid = userInfo.CHARACTER_CODE_SET_ID;
+                }
+
+                // if the job ccsid is *SYSVAL, then assign it to sysval
+                if (this.userJobCcsid === CCSID_SYSVAL) {
+                  this.userJobCcsid = this.qccsid;
+                }
+
+                // Let's also get the user's default CCSID
+                try {
+                  const [activeJob] = await this.runSQL(`Select DEFAULT_CCSID From Table(QSYS2.ACTIVE_JOB_INFO( JOB_NAME_FILTER => '*', DETAILED_INFO => 'ALL' ))`);
+                  this.userDefaultCCSID = Number(activeJob.DEFAULT_CCSID);
+                }
+                catch (error) {
+                  const [defaultCCSID] = (await this.runCommand({ command: "DSPJOB OPTION(*DFNA)" }))
+                    .stdout
+                    .split("\n")
+                    .filter(line => line.includes("DFTCCSID"));
+
+                  const defaultCCSCID = Number(defaultCCSID.split("DFTCCSID").at(1)?.trim());
+                  if (defaultCCSCID && !isNaN(defaultCCSCID)) {
+                    this.userDefaultCCSID = defaultCCSCID;
+                  }
+                }
+
+              } catch (e) {
+                // Oh well!
+                console.log(e);
+              }
+            }
+
+            let userCcsidNeedsFixing = false;
+            let sshdCcsidMismatch = false;
+
+            if (this.getComponent(cqsh)) {
+              // If cqsh is available, but the user profile CCSID is bad, then cqsh won't work
+              if (this.getCcsid() === CCSID_NOCONVERSION) {
+                userCcsidNeedsFixing = true;
+              }
+            }
+            
+            if (!this.canUseCqsh) {
```

Comment:
With the above change:
```suggestion
            else {
```